src/amo/pages/AddonVersions/index.js

```diff
@@ -0,0 +1,210 @@
+} from 'core/types/router';
+import type { I18nType } from 'core/types/i18n';
+
+import './styles.scss';
+
+type Props = {|
+  location: ReactRouterLocationType,
+|};
+
+type InternalProps = {|
+  ...Props,
+  addon: AddonType | null,
+  addonIsLoading: boolean,
+  areVersionsLoading: boolean,
+  clientApp: string,
+  dispatch: DispatchFunc,
+  errorHandler: ErrorHandlerType,
+  history: ReactRouterHistoryType,
+  i18n: I18nType,
+  lang: string,
+  match: {|
+    ...ReactRouterMatchType,
+    params: {
+      slug: string,
+    },
+  |},
+  versions?: Array<AddonVersionType>,
+|};
+
+export class AddonVersionsBase extends React.Component<InternalProps> {
+  constructor(props: InternalProps) {
+    super(props);
+
+    this.loadDataIfNeeded();
+  }
+
+  componentDidUpdate(prevProps: InternalProps) {
+    this.loadDataIfNeeded(prevProps);
+  }
+
+  loadDataIfNeeded(prevProps?: InternalProps) {
+    const {
+      addon,
+      addonIsLoading,
+      areVersionsLoading,
+      dispatch,
+      errorHandler,
+      versions,
+      match: {
+        params: { slug },
+      },
+    } = this.props;
+
+    const oldAddon = prevProps && prevProps.addon;
+
+    if (errorHandler.hasError()) {
+      log.warn('Not loading data because of an error');
+      return;
+    }
+
+    // Fetch versions when the add-on changes.
+    const addonHasChanged = oldAddon && oldAddon.slug !== slug;
+
+    if ((!addon || addonHasChanged) && !addonIsLoading) {
+      dispatch(fetchAddon({ slug, errorHandler }));
+    }
+
+    if (addon && !areVersionsLoading && (!versions || addonHasChanged)) {
+      dispatch(
+        fetchVersions({
+          errorHandlerId: errorHandler.id,
+          slug,
+        }),
+      );
+    }
+  }
+
+  render() {
+    const { addon, errorHandler, i18n, versions } = this.props;
+
+    let latestVersion;
+    let olderVersions = [];
+    if (Array.isArray(versions)) {
+      latestVersion = versions.length ? versions[0] : null;
+      olderVersions = versions.slice(1);
+    }
+
+    let header = '';
+    if (addon && versions) {
+      header = i18n.sprintf(
+        i18n.ngettext(
+          '%(addonName)s Version history - %(total)s version',
+          '%(addonName)s Version history - %(total)s versions',
+          versions.length,
+        ),
+        {
+          addonName: addon.name,
+          total: i18n.formatNumber(versions.length),
+        },
+      );
+    }
+
+    return (
+      <div className="AddonVersions">
+        {addon && (
+          <Helmet>
+            <title>{header}</title>
+          </Helmet>
+        )}
+
+        {errorHandler.renderErrorIfPresent()}
+
+        <AddonSummaryCard addon={addon} headerText={header} />
+
+        <CardList
+          className="AddonVersions-versions"
+          header={header || <LoadingText />}
+        >
+          <ul>
+            <AddonVersionCard
+              addon={addon}
+              headerText={i18n.gettext('Latest version')}
+              key="latestVersion"
```

Comment:
This could use `latestVersion.id` for consistency

Reply:
I started with that, but `latestVersion` can also be `null` and `undefined` here, so I cannot just use `latestVersion.id`. I could throw in an `||`, but at that point it seems to make sense to just use the hardcoded value `"latestVersion"`, which is guaranteed to be unique from all the other `AddonVersionCard`s.